scripts/lint_jobs.sh

```diff
@@ -0,0 +1,175 @@
+set -e
```

Comment:
I think this test is great, I'd suggest implementing it in Go so you can use the same [parsing](https://github.com/kubernetes/test-infra/blob/59e70a3d6443f37ba94961c9cef567655431bc20/prow/config/config.go#L1091) and [types](https://pkg.go.dev/k8s.io/test-infra/prow/config#JobConfig) as Prow for more advanced (and testable) checks in the future.

I'd also be great to run [`checkconfig`](https://github.com/kubernetes/test-infra/tree/2abb7ff26579325f6a335990003f665755c96d7a/prow/cmd/checkconfig) before this test for basic schema validation as well